linux/src/async_socket_linux.c

```diff
@@ -215,6 +220,9 @@ static int thread_worker_func(void* parameter)
             {
                 // Codes_SRS_ASYNC_SOCKET_LINUX_11_082: [ thread_worker_func shall receive the ASYNC_SOCKET_RECV_CONTEXT value from the ptr variable from the epoll_event data ptr. ]
                 ASYNC_SOCKET_RECV_CONTEXT* recv_context = events[index].data.ptr;
+                #ifdef USE_VALGRIND
+                    ANNOTATE_HAPPENS_AFTER(recv_context);
+                #endif
                 if (recv_context != NULL)
```

Comment:
>recv_context != NULL [](http://example.com/codeflow?start=20&length=20)

can this ever be `NULL`? If not, then why check it? 

Also noticing a lack of `else` here...

Reply:
Same in line 252

Reply:
I believe that you're correct it would not be NULL